### ISSUE TITLE
[mixed time-series; time-series bar chart v2] Bar chart jumps to very narrow bars in larger time ranges

### ISSUE THREAD

**Issue by rumbin** [apache/superset#18232]
Depending on the displayed time range, the width of the bar-chart bars becomes very narrow, up to the point where the bars become nearly invisible.

The effect depends on the selected time grain, but I haven't found a pattern yet.

**I can replicate this on both, mixed time-series _and_ time-series bar chart v2.**

#### How to reproduce the bug

1. Create a diagram of type **mixed time-series** (bar chart here!) or ** time-series bar chart v2**, based on a dataset that has at least 1 year of data in it
2. Select a time range of 1 year
3. Vary the time grain between weeks and quaters
4. Activate _Data Zoom_ and play with it; alternatively reduce the time range gradually

### Expected results

The bars of the bar chart should continuously be varied according to the time grain and the time rage, so the bar-to-gap ratio is harmonic. 

The old time-series bar chart may serve as a good example here.

### Actual results

The width of the bars is harmonically scaled in some time ranges, but when a certain threshold is exceeded, the bar width jumps to nearly invisible vertical lines.

The odd thing here is, that I cannot identify a pattern here:

* For **weekly** time-grain the jump is at about end of march, when the whole year is displayed.
* For **monthly** time-grain the threshold is at the beginning of march. So far so good.
* But: for **quaterly** time_grain the bars are _always_ narrow lines, although there is by far enough space now.

#### Screenshots

The first gif shows that the threshold where the bars jump from narrow to wide and back can be narrowed by both _Data Zoom_ and by changing the time range:

![bar_chart_width_jump_weekly](https://user-images.githubusercontent.com/1220356/151816596-109a5d4a-bd2f-417c-81a6-0d89676a9ba0.gif)

The second gif shows that the threshold depends on the time grain, where the few quaterly bars behave most erratic:

![bar_chart_width_jump_time_grain](https://user-images.githubusercontent.com/1220356/151817428-6caf1947-81ce-485f-81aa-8cdc5650793d.gif)

### Environment

- browser type and version: Chrome  96.0.4664.93
- superset version: 1.4.0
- all other versions: as defined in Superset's Docker image
- Feature flags:

```
# stable:
      "THUMBNAILS": True,
      "SQLLAB_BACKEND_PERSISTENCE": True,
      "ENABLE_TEMPLATE_PROCESSING": True,
      "DASHBOARD_CROSS_FILTERS": True,

      # experimental:
      "ALERT_REPORTS": True,
      "ALERTS_ATTACH_REPORTS": True,
      "DASHBOARD_NATIVE_FILTERS": True,

      # unclassified:
      "ENABLE_EXPLORE_DRAG_AND_DROP": True,
      "ENABLE_DND_WITH_CLICK_UX": True,

      # development:
      "DASHBOARD_NATIVE_FILTERS_SET": True
```

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by rumbin** [comment]
This issue should not be closed by the stale bot, in my eyes.

**Comment by rusackas** [comment]
@rumbin I don't want to dismissively close this, but it's been upward of a year, and I _believe_ this was solved by a few correlated PRs. Can you verify this is resolved (or not!) in 2.0.1?

**Comment by rumbin** [comment]
I cannot reproduce this issue any longer on v2.0.1 with Bar Chart v2 and custom x-axis enabled.
Closing it, @rusackas.
